### DISCUSSION TITLE
Adding a computer to iCub cluster

### DISCUSSION THREAD

**Discussion by paliasgh** [robotology/community#719] [Q&A] [answered]
Hello,

We ran into a problem when adding a new computer to our system. We followed the procedure described [here](https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/generic-machine/) for setting up a generic iCub environment on the new machine (Ubuntu 20.04). The problem we could not fix is that NFS share directories (`/usr/local/src/robot` and `/home/icub/.local/share/yarp') are not editable from the client's side. This means the newly set up computer can see the content of these locations mounted from the server, but can not edit anything or add anything to it. I would appreciate any ideas on how this might have been caused.

I checked the server side and the export configurations were as defined [here](https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/icub-server-from-scratch/#nfs-server) (other computers work fine, i.e., can edit these locations):

```
/exports/code           10.0.0.0/255.255.255.0(rw,sync,no_subtree_check)
/exports/local_yarp     10.0.0.0/255.255.255.0(rw,sync,no_subtree_check)
```

Thank you very much!

**Comment by pattacini** [comment, marked as answer]
Hi @paliasgh 

Just talked with @davidelasagna and we think that the issue may be related to the [UID](https://linuxhandbook.com/uid-linux/) of the user `icub`.

Can you verify that the UID of `icub` is the same on all the machines of your cluster?
Just do:
```console
id icub
```

To properly work with NFS, we ought to make sure that `icub` has the same UID on all the systems.
If you discover that on your new computer the UID is different, you can re-assign the ID as explained in the how-to guide I linked above.

Essentially, launch as root
```
usermod -u 1000 icub
```

being 1000 the expected UID.

> [!caution]
> There's a caveat though as reported in the guide:
>
>>When you update the UID of a user, what happens to the files owned by this user?While all the files in the home directory of user_2 will have their associated UID changed, you’ll have to manually update the associated UID of other files outside the home directory.
>>
>>What you can do is manually update the ownership of the files associated with the old UID of the user_2.
>> `find / -user old_uid_of_user_2 -exec chown -h user_2 {} \;`

**Reply by pattacini** [reply]
@paliasgh 

Remember to verify that the UID 1000 (or the one used on the server) is freely available on the new PC, meaning is not already associated with another user. To this end, inspect the content of the `/etc/passwd` file, as per the guide.

If the UID has already been taken, then you would need to delete that user (if you don't need it) or move it to a different UID first.

**Reply by paliasgh** [reply]
Hello @pattacini , thank you very much for your prompt answer. yes! That was the issue. In fact, the UID of all computers in our cluster was 1001. I change the new one to 1001 and that fixed the issue.
